PARCEL_2_RFC.md

```diff
@@ -0,0 +1,1285 @@
+else (ex. `--mode production` or `--mode development`). Defaults to the
+`NODE_ENV` environment variable.
+
+##### `--development, --dev`
+
+Aliases for `--mode development`.
+
+##### `--production, --prod`
+
+Aliases for `--mode production`.
+
+##### `--test`
+
+Aliases for `--mode test`.
+
+#### `--public-url <url>`
+
+[todo]
+
+#### `--log-level <level>`
+
+Set the log level, either "0" (no output), "1" (errors), "2" (warnings +
+errors) or "3" (all). (Default: 2)
+
+#### `--version, -v, -V`
+
+Return the current version of Parcel.
+
+#### `--help, -h`
+
+Get help with the CLI.
+
+## Parcel Config
+
+Parcel has always and will always work out of the box for many projects with
+zero configuration. It should always be extremely simple to get started. But if
+you do want more control, we give you the tools to do so.
+
+### Configuring external tools
+
+A huge part of what Parcel does is run other tools over your code. Instead of
+pulling all that configuration into Parcel, we make use of their own
+configuration systems. So if you're using Babel, you should just use `.babelrc`
+files to configure it.
+
+When we do need to introduce config, we create tool specific config files in
+order to do so.
+
+### Configuring Parcel
+
+When you do need to configure Parcel, it will be in one of 3 places.
+
+- If you need to configure the CLI, it will be a CLI flag
+- If you need to configure your package, it will be in the `package.json`
+- If you need to configure something with your files or the Parcel asset
+  pipeline, it will be in `.parcelrc`
+
+### `package.json`
+
+[todo]
+
+```json
+{
+  "name": "foo",
+  "main": "dist/main/index.js",
+  "module": "dist/module/index.js",
+  "browser": "dist/browser/index.js",
+  "browserslist": ["> 1%", "not dead"],
+  "engines": {
+    "node": ">=4.x"
+  },
+  "source": "src/index.js",
+  "targets": {
+    "main": {
+      "node": ["^4.0.0"]
+    },
+    "module": {
+      "node": ["^8.0.0"]
+    },
+    "browser": {
+      "browsers": ["> 1%", "not dead"]
+    }
+  },
+  "alias": {
+    "react": "preact-compat",
+    "react-dom": "preact-compat"
+  }
+}
+```
+
+#### `package.json#name`
+
+**(Required)** The name of the package is always required in order to be
+considered a valid `package.json`.
+
+```json
+{
+  "name": "my-package"
+}
+```
+
+#### `package.json#version`
+
+**(Required)** All packages inside `node_modules` must have a `package.json#version`.
+
+```json
+{
+  "version": "1.0.0"
+}
+```
+
+#### `package.json#main`
+
+**(Required)** This is the "main" target's entry point for the package.
+
+```json
+{
+  "main": "dist/main/index.js"
+}
+```
+
+See [Targets](#targets)
+
+#### `package.json#module`
+
+This is the "module" target's entry point for the package.
+
+```json
+{
+  "module": "dist/module/index.js"
+}
+```
+
+See [Targets](#targets)
+
+#### `package.json#browser`
+
+This is the "browser" target's entry point for the package.
+
+```json
+{
+  "browser": "distflinesof/browser/index.js"
```

Comment:
I think this is a typo. Shouldn't it be `dist/browser/index.js`?

Reply:
...indeed, wtf happened?